src/store/reader.rs

```diff
@@ -128,6 +128,102 @@ impl StoreReader {
         })
     }
 
+    /// Iterator over all Documents in their order as they are stored in the doc store.
+    /// Use this, if you want to extract all Documents from the doc store.
+    /// The delete_bitset has to be forwarded from the `SegmentReader` or the results maybe wrong.
+    pub fn iter<'a: 'b, 'b>(
+        &'b self,
+        delete_bitset: Option<&'a DeleteBitSet>,
+    ) -> impl Iterator<Item = crate::Result<Document>> + 'b {
+        self.iter_raw(delete_bitset).map(|raw_doc| {
+            let raw_doc = raw_doc?;
+            let mut cursor = raw_doc.get_bytes();
+            Ok(Document::deserialize(&mut cursor)?)
+        })
+    }
+
+    /// Iterator over all RawDocuments in their order as they are stored in the doc store.
+    /// Use this, if you want to extract all Documents from the doc store.
+    /// The delete_bitset has to be forwarded from the `SegmentReader` or the results maybe wrong.
+    pub fn iter_raw<'a: 'b, 'b>(
+        &'b self,
+        delete_bitset: Option<&'a DeleteBitSet>,
+    ) -> impl Iterator<Item = crate::Result<RawDocument>> + 'b {
+        let last_docid = self
+            .block_checkpoints()
+            .last()
+            .map(|checkpoint| checkpoint.doc_range.end)
+            .unwrap_or(0);
+        let mut checkpoint_block_iter = self.block_checkpoints();
+        let mut curr_checkpoint = checkpoint_block_iter.next();
+        let mut curr_block = curr_checkpoint
+            .as_ref()
+            .map(|checkpoint| self.read_block(&checkpoint));
+        let mut block_start_pos = 0;
+        let mut num_skipped = 0;
+        (0..last_docid)
+            .filter_map(move |doc_id| {
+                // filter_map is only used to resolve lifetime issues between the two closures on
```

Comment:
I am incapable of reading this :)

But I think this is ok considering it has no adherence outside of this function